tf2_ros/include/tf2_ros/message_filter.h

```diff
@@ -423,12 +432,17 @@ class MessageFilter : public MessageFilterBase, public message_filters::SimpleFi
       const auto & handle = std::get<0>(param);
       const auto & stamp = std::get<1>(param);
       const auto & target_frame = std::get<2>(param);
-      buffer_.waitForTransform(
+      tf2_ros::TransformStampedFuture future = buffer_.waitForTransform(
         target_frame,
         frame_id,
         stamp,
         buffer_timeout_,
         std::bind(&MessageFilter::transformReadyCallback, this, std::placeholders::_1, handle));
+
+      if (0 != future.getHandle() || 0xffffffffffffffffULL != future.getHandle()) {
```

Comment:
Does `0xffffffffffffffffULL` represent the maximum `uint64_t` value? If so, wouldn't `std::numeric_limits<uint64_t>::max()` be better?

Reply:
Yes and no.  While it *is* the maximum value, we also use it as a hard-coded constant elsewhere, like https://github.com/ros2/geometry2/blob/7a660094d0da9c463be5fea1df60d836b3349e9b/tf2_ros/src/buffer.cpp#L249 .  It is not ideal, and maybe we should instead switch to using the limit, but I think that would be a separate patch.

Reply:
I think it is worthwhile to add a comment here that if the future is `0` or `0xffffffffffffffffULL`, then `waitForTransform` will have already called the callback.  It's certainly possible to discover this by reading the Buffer/BufferCore code, but it is not super-obvious.